stdlib/public/Concurrency/Task.swift

```diff
@@ -0,0 +1,273 @@
+////===----------------------------------------------------------------------===//
+////
+//// This source file is part of the Swift.org open source project
+////
+//// Copyright (c) 2020 Apple Inc. and the Swift project authors
+//// Licensed under Apache License v2.0 with Runtime Library Exception
+////
+//// See https://swift.org/LICENSE.txt for license information
+//// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+////
+////===----------------------------------------------------------------------===//
+
+import Swift
+@_implementationOnly import _SwiftConcurrencyShims
+
+// ==== Task -------------------------------------------------------------------
+
+/// An asynchronous task (just "Task" hereafter) is the analogue of a thread for
+/// asynchronous functions. All asynchronous functions run as part of some task.
+///
+/// A task's execution can be seen as a series of periods where the task was
+/// running. Each such period ends at a suspension point or -- finally -- the
+/// completion of the task.
+///
+/// These partial periods towards the task's completion are `PartialAsyncTask`.
+/// Partial tasks are generally not interacted with by end-users directly,
+/// unless implementing a scheduler.
+public struct Task {
+}
+
+// ==== Current Task -----------------------------------------------------------
+
+extension Task {
+  /// Returns the currently executing `Task`.
+  ///
+  /// As invoking this function is only possible from an asynchronous context
+  /// it is always able to return the current `Task` in which we are currently
+  /// running.
+  public static func current() async -> Task {
+    fatalError("\(#function) not implemented yet.")
```

Comment:
This will need a compiler builtin that knows about the `async` calling convention. We'll deal with that separately.

Reply:
Right 👍